File: infrastruktur/src/main/java/com/github/haschi/haushaltsbuch/infrastruktur/VertxCommandGateway.java
package com.github.haschi.haushaltsbuch.infrastruktur;

import org.axonframework.messaging.annotation.MetaDataValue;

import java.util.concurrent.CompletableFuture;

public interface VertxCommandGateway
{
    <T> CompletableFuture<T> send(
            Object anweisung,
            @MetaDataValue("threadId") final long threadId);
}

File: infrastruktur/src/main/kotlin/com/github/haschi/haushaltsbuch/infrastruktur/VertxCommandGateway.kt
package com.github.haschi.haushaltsbuch.infrastruktur

import org.axonframework.messaging.annotation.MetaDataValue

import java.util.concurrent.CompletableFuture

interface VertxCommandGateway
{
    fun <T> send(
            anweisung: Any,
            @MetaDataValue("threadId") threadId: Long): CompletableFuture<T>
}
